src/gui/qgsquerybuilder.cpp

```diff
@@ -94,6 +96,48 @@ QgsQueryBuilder::QgsQueryBuilder( QgsVectorLayer *layer,
   layerSubsetStringChanged();
 
   lblDataUri->setText( tr( "Set provider filter on %1" ).arg( layer->name() ) );
+
+  lblProviderStorageType->setOpenExternalLinks(true);
+
+  //TODO what to put there for other providers, e.g. postgis?
+  QStringList supportedSqlDialectsText = QStringList();
+  if ( layer->dataProvider()->name() == "ogr" ) {
+
+    GDALDriverH driver = GDALGetDriverByName( layer->dataProvider()->storageType().toLocal8Bit().constData() );
+    const char* supportedSqlDialects = GDALGetMetadataItem( driver, GDAL_DMD_SUPPORTED_SQL_DIALECTS, nullptr );
+
+    QStringList supportedSqlDialectsList = QString(supportedSqlDialects).split(" ");
+    int count = supportedSqlDialectsList.count();
+    for( int i=0; i<count; i++ )
+    {
+      QString supportedSqlDialect = supportedSqlDialectsList.at(i);
+      QString link = QString();
+      QString title = supportedSqlDialect;
+      if ( supportedSqlDialect == "NATIVE" ) {
```

Comment:
```suggestion
      if ( supportedSqlDialect == QStringLiteral( "NATIVE" ) ) {
```
and other below ...

Reply:
Cheers! Luckily obsoleted by using Nyall's new methods instead :)